package.json

```
@@ -67,6 +65,11 @@
     "typedoc": "0.22.11",
     "typescript": "4.5.5"
   },
+  "peerDependencies": {
+    "@types/tough-cookie": ">=4.0.1",
+    "extend": ">=3.0.2",
+    "tough-cookie": ">=4.0.0"
```

I am a little concerned about these having no max. In the tough-cookie case, at least, a change of major version is likely to break the session auth patch.

Since we are relying on the core installs for these dependencies I think we should match their ranges:

```suggestion
    "@types/tough-cookie": "^4.0.0",
    "extend": "^3.0.2",
    "tough-cookie": "^4.0.0"
```

Noting the core asks for `@types/tough-cookie` `^4.0.0`, not sure if we have as specific need for `^4.0.1`.

---

> of major version is likely to break the session auth patch

I see, I was hesitating between these 2 formats. I ended up with the `>=` range because if core jumps one major version, we have to do the update too. Wdyt of using carets for `tough-cookie` and `@types/tough-cookie` and keep the range for `extend`?

> Noting the core asks for `@types/tough-cookie` `^4.0.0`, not sure if we have as specific need for `^4.0.1`.

Yeah, here I used `npm list @types/tough-cookie`, so I did not copy the defined versions in core, but tried to use the latest possible version everywhere instead.

---

>if core jumps one major version, we have to do the update too.

Yeah, I wish there was a way for to say "use the version from this specific dependency", but as it stands there is no guarantee where we will pick up the peer from if it is there. My concern is that some _other_ dependency might bring one of the packages at a higher major than the core uses, in which case we might break even if core didn't change major.

When the core itself makes a major change I think there is still a risk of breaking with either approach, but our use of a pinned core version should at least mean we observe breaks in the PRs to update the core version. The downside of the `^` approach is that in NPM 7+ we will get an old peer installed for us if the core change their rane and we didn't change ours to match. We'd still see if this caused a break though in the version bump PR and I think it is less risky than the "other dependency" problem.

>Wdyt of using carets for tough-cookie and @types/tough-cookie and keep the range for extend?

What is the motivation for splitting? Both packages are very widely used (extend 20M downloads a week, tough-cookie 40M) and provide a risk of another dependency bringing a higher version. Granted I think `extend` is probably less likely to have a major version change given most of the functionality is provided natively in Node now and it hasn't had a release in 4 years.

>so I did not copy the defined versions in core

Our intention is to use the packages from core, so I think we should match exactly the versions they declare as it gives us the best chance of getting the version we want/expect. I agree maintaining that match is likely to be a little burdensome, but hopefully neither package will be jumping major very much.

---

> What is the motivation for splitting?

I was just reacting to this: 

> In the tough-cookie case, at least, a change of major version is likely to break the session auth patch.

-------------
> Our intention is to use the packages from core, so I think we should match exactly the versions they declare as it gives us the best chance of getting the version we want/expect.

Yes, but they declared `^4.0.0` which is translated now as `4.0.1`. So now we are pulling `4.0.1` anyway, and if we allowed dependabot to update this, it should be `4.0.1` now. I hope dependabot will pull the appropriate updates for peerDependencies in the future.

---------
> maintaining that match is likely to be a little burdensome

IIUC this is good for us: https://github.com/dependabot/feedback/issues/54#issuecomment-442901332

------
> I wish there was a way for to say "use the version from this specific dependency"

I was looking for a similar option, I found `peerDependenciesMeta`, but that is only good for avoiding warning messages, so that does not help for us in this case.

----
all in all: I rebased and addressed your comments: [c7918fa](https://github.com/IBM/cloudant-node-sdk/pull/712/commits/c7918fa22891229f716b195b5ab62033db6fab04)